Library/Homebrew/extend/os/mac/requirements/x11_requirement.rb

```diff
@@ -0,0 +1,44 @@
+require "requirement"
+
+class XQuartzRequirement < Requirement
+  include Comparable
+
+  fatal true
+  cask "xquartz"
+  download "https://xquartz.macosforge.org"
+
+  env { ENV.x11 }
+
+  def initialize(name = "x11", tags = [])
+    @name = name
+    # no-op on version specified as a tag argument
```

Comment:
Let's do the same thing on the generic X11 requirement; they should have the same interface. I think the only thing they should probably differ on is the `satisfy` and `min_version` stuff (and would be good to have a `min_version` method there too).

Reply:
These two `initialize` functions are identical. Can you please elaborate?

Generic: https://github.com/sjackman/homebrew/blob/9f7278899e24fa3567db7e15050a63e8f3242679/Library/Homebrew/requirements/x11_requirement.rb#L8

macOS: https://github.com/sjackman/homebrew/blob/9f7278899e24fa3567db7e15050a63e8f3242679/Library/Homebrew/extend/os/mac/requirements/x11_requirement.rb#L12

Reply:
Apologies, I misread this.